File: back-end/post-service/src/main/java/com/muates/postservice/model/dto/response/PostResponse.java
package com.muates.postservice.model.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PostResponse {

    private Long id;
    private Long userId;
    private String content;
    private Date createdDate;
    private Date updatedDate;
    private List<PostMediaResponse> media;
    private List<PostLikeResponse> likes;
    private List<PostCommentResponse> comments;
    private String username;
    private String profilePicture;
}

File: front-end/src/app/core/models/post/post-response.ts
import { PostCommentResponse } from "./post-comment-response";
import { PostLikeResponse } from "./post-like-response";
import { PostMediaResponse } from "./post-media-response";

export class PostResponse {
    id!: number;
    userId!: number;
    content!: string;
    createdDate!: Date;
    updatedDate!: Date;
    media!: PostMediaResponse[];
    likes!: PostLikeResponse[];
    comments!: PostCommentResponse[];
    username!: string;
    profilePicture!: string;
}
